core/tools/utils.py

```diff
@@ -99,6 +102,18 @@ def sleep_forever():
         time.sleep(60)
 
 
+def send_update_to_api(task_uuid: str, status: str, routing_key: str, payload: str):
+    connection_params = pika.ConnectionParameters(host="rabbitmq", port=5672)
+    connection = pika.BlockingConnection(connection_params)
+    channel = connection.channel()
+    channel.exchange_declare(exchange="status", exchange_type="direct")
+    message = json.dumps({"task_uuid": task_uuid, "status": status, "payload": payload})
+    channel.basic_publish(exchange="", routing_key=routing_key, body=message)
+    print("status sent")
+    channel.close()
+    connection.close()
```

Comment:
I will move this function to a summarizer since it is the only worker using it. Also, I read that you’re right. It’s better practice to leave the connection open, but leaving the channel open will still consume memory. If they aren’t used, they should be closed. However, I am not sure if the time between using them will be long enough that closing and reopening them will be beneficial.